index_put_ for complex tensors on CUDA

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51148 index_put_ for complex tensors on CUDA**

Differential Revision: [D26102025](https://our.internmc.facebook.com/intern/diff/D26102025)

anjali411 committed:
index_put_ for complex tensors on CUDA

[ghstack-poisoned]

anjali411 committed:
Update on "index_put_ for complex tensors on CUDA"




[ghstack-poisoned]

albanD commented on aten/src/ATen/native/cuda/Indexing.cu:
```diff
@@ -230,7 +230,7 @@ void index_put_accum_kernel(Tensor & self, const c10::List<c10::optional<Tensor>
       dim3 block(C10_WARP_SIZE, indices_per_block);
 
-      AT_DISPATCH_ALL_TYPES_AND3(at::ScalarType::Half, at::ScalarType::Bool, at::ScalarType::BFloat16,
+      AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(at::ScalarType::Half, at::ScalarType::Bool, at::ScalarType::BFloat16,
```
Is cdouble included in this case? Should cdouble be included in the tests as well?

anjali411 replied:
yeah it is! sure I could include that in the test as well

anjali411 replied:
@albanD done!

anjali411 committed:
Update on "index_put_ for complex tensors on CUDA"


Differential Revision: [D26102025](https://our.internmc.facebook.com/intern/diff/D26102025)

[ghstack-poisoned]

albanD reviewed (commented):
LGTM thanks for the update!
